### DISCUSSION TITLE
date in filter for mysql table as data provider

### DISCUSSION THREAD

**Discussion by stephlegrand** [Budibase/budibase#3684] [Help]
Hello,

This is something that used to work until the last updates, but maybe I am doing it wrong.

I have a database with events. Each event has a **start date** and and an **end_date**:
![image](https://user-images.githubusercontent.com/50490522/144744927-d31c5ba5-6b9c-4358-bb39-5fdd1d9e499e.png)

In a form, I want to offer an event picker, but only for events that are still happening or coming up. So I have a data provider with the events table, and a filter on the end_date:
![image](https://user-images.githubusercontent.com/50490522/144744913-b7c14f36-7414-46a7-ae9f-b65aae4fe487.png)

This used to work fine, but since the latest updates (v1.0.x last week) the list is just empty, so I am assuming that I am not using the {{ date }} binding correctly. I have looked everywhere for instructions, but I am at a loss. Does anyone have an idea?

**Comment by mjashanks** [comment]
Hi Steph. I think the binding you are looking for is `{{ now }}`. Have checked, and this works for me.

**Reply by stephlegrand** [reply]
Hi Michael,

I have tried {{ date }}, {{ now }} and {{ date now }} but nothing works.

If I remove filters, I get the full list as expected:

![image](https://user-images.githubusercontent.com/50490522/144803025-49679a79-3832-4bf6-9d18-cf660ceb3afc.png)


But with the filter I should get 2022 events, and I just get nothing:

![image](https://user-images.githubusercontent.com/50490522/144803319-0d012ea2-088c-4669-b1b7-26979f8161f2.png)

**Reply by mjashanks** [reply]
I have set myself up the same as above, and can't reproduce this.
```
> Form
  > Data Provider - date filter > {{ now }}
    > Options Picker - using data provider as source.
```

It filters correctly.

One thing that I noticed - I initially set my `Value Column` to a column that had an undefined value in it. The items did not appear for me. Once I set the value column to a column that had values (e.g. `_id`), the items reappeared. Could this be the issue?

**Reply by stephlegrand** [reply]
Thanks for looking into it :)

This is not it - my `Value Column` is `suitecrm_id`, which has a value for every event.
